### PR TITLE
DDD-79 Create user info update endpoint

### PR DESCRIPTION
**Pull request by KornelisMax**
Added possibility to update user info inside user's profile.

Created interface for UserInfoDTO (structure seems to be the same for POST/PUT actions)

@egisxxegis @Gintasp @Accir @Zygis0321

### CODE REVIEW

File: src/main/java/lt/vu/application/user/service/UpdateUserInfoService.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import lt.vu.application.address.factory.AddressFactory;
+import lt.vu.persistence.orm.entities.User;
+import lt.vu.persistence.orm.repository.UserRepository;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class UpdateUserInfoService {
```
Comment (KornelisMax) [inline src/main/java/lt/vu/application/user/service/UpdateUserInfoService.java]: Not sure about the naming; Put or Update?

Comment (giplonis) [reply src/main/java/lt/vu/application/user/service/UpdateUserInfoService.java]: Put is a low-level (http) implemention detail. A service class is a high level implementation. So in this case, `update` is the better word. Also, I think it's more clear when services start with model/domain name, not a verb - `UserInfoUpdateService` - a service that updates UserInfo.

File: src/main/java/lt/vu/application/user/service/UpdateUserInfoService.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public void updateUser(User user, UserInfoDTO userInfoDTO){
+        user.setEmail(userInfoDTO.getEmail());
+        user.setFirstName(userInfoDTO.getFirstName());
+        user.setLastName(userInfoDTO.getLastName());
+        user.setPhoneNumber(userInfoDTO.getPhoneNumber());
+        user.setAddress(this.addressFactory.createFromDTO(userInfoDTO.getAddress()));
+
+        this.userRepository.update(user);
+    }
+
```
Comment (giplonis) [inline src/main/java/lt/vu/application/user/service/UpdateUserInfoService.java]: Unintentional blank line? 😄

File: src/main/java/lt/vu/web/api/v1/dto/user/UpdateUserDTO.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import lt.vu.web.api.v1.dto.address.PostAddressDTO;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import javax.ws.rs.FormParam;
+
+@Getter @Setter
+public class UpdateUserDTO implements UserInfoDTO {
```
Comment (giplonis) [inline src/main/java/lt/vu/web/api/v1/dto/user/UpdateUserDTO.java]: I think we should stick to the usual way we name DTOs - `<HTTP method name><model name>DTO`. In this case - `PutUserInfoDTO`.

File: src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    ),
+            }
+    )
+    public Response putAction(
+            @RequestBody(
+                    required = true,
+                    content = @Content(schema = @Schema(implementation = UpdateUserDTO.class))
+            ) @Valid UpdateUserDTO updateUserInfoDTO) throws BadRequestException, NotFoundException {
+
+        updateUserInfoService.updateUser(this.user, updateUserInfoDTO);
```
Comment (giplonis) [inline src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java]: `this.updateUserInfoService` would look more consistent 😄

File: src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    private UpdateUserInfoService updateUserInfoService;
+
+    @PUT
+    @Path("/")
+    @Authorized
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Transactional
+    @Operation(
+            summary = "Change user's data",
```
Comment (giplonis) [inline src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java]: I know IntelliJ does this, but for consistency, we have 4 spaces (1 tab) indentation in other files.

File: src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/user/details")
```
Comment (giplonis) [inline src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java]: Why not `/user-info` (the same as entity name)?

Comment (KornelisMax) [reply src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java]: Good point.

File: src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java (hunk 5 above)
Comment (KornelisMax) [reply src/main/java/lt/vu/web/api/v1/controller/user/UpdateUserInfoController.java]: Yes.. Didn't notice that :D Linter would be great :D

File: src/main/java/lt/vu/application/user/service/UpdateUserInfoService.java (hunk 2 above)
Comment (KornelisMax) [reply src/main/java/lt/vu/application/user/service/UpdateUserInfoService.java]: Yes... 🙈

**Review by giplonis — APPROVED** [review]
🚀

**Pull request merged**